### PR TITLE
Codegen rework

### PR DESCRIPTION
**Pull request by lukas-valenta**
## Description

- codegen is changed to produce .js and .d.ts files

## Motivation and Context

- it is needed to use the generated types from both typescript and javascript projects

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/logic/install.ts [hunk 1]
```diff
@@ -204,28 +204,26 @@ export async function resolveInstallationRequests(
     }
   }
 
+  await generateTypes(phase3, superJson);
```
Comment (TheEdward162) [inline src/logic/install.ts:207]: Can I has the doc on this function updated? With like `5. generate typings` or something

File: src/logic/install.test.ts [hunk 2]
```diff
@@ -527,29 +526,14 @@ describe('Install CLI logic', () => {
...
-        expect(fetchProfileAST).toHaveBeenNthCalledWith(
-          1,
-          'starwars/first@1.0.0'
-        );
-        expect(fetchProfileAST).toHaveBeenNthCalledWith(
-          2,
-          'starwars/second@2.0.0'
-        );
-
-        expect(mockWrite).toHaveBeenCalledTimes(5);
```
Comment (TheEdward162) [inline src/logic/install.test.ts:552]: but why?

File: src/logic/install.ts (hunk 1 above)
Comment (lukas-valenta) [reply src/logic/install.ts:207]: what value will that add?

Comment (TheEdward162) [reply src/logic/install.ts:208]: Inner peace and not having to read the code when you just want to see where the typings are generated

Comment (TheEdward162) [reply src/logic/install.ts:208]: I mainly meant the enclosing function (line 140) 🙏

**Review by TheEdward162 — APPROVED** [review]
is ok but it'd be nice if you updated the doc comment 🥇

**Pull request merged**